events/spamdetect.js

```diff
@@ -0,0 +1,85 @@
+// how many identical messages must be sent in a row for Wilson to take action
+const MESSAGES_TO_CACHE = 3
+// how long messages should be remembered at least (it will take between 1x and 2x this duration before they are removed from memory)
+const PURGE_INTERVAL_MINUTES = 3
+// this gets sent to the user whose messages have been detected as spam
+const MESSAGE_TO_USER = `As a spam prevention measure, you've been muted for sending the exact same message ${MESSAGES_TO_CACHE} times in a row. Please contact a moderator to be un-muted.`
+// messages in these channels are ignored
+const IGNORED_CHANNELS = [
+    '649186872026988557', // #botspam
+    '649186919317504012', // #random
+]
+// only check messages in these guilds
+const ACTIVE_GUILDS = [
+    '649165975647682560', // Wintergatan
+]
+const REPORT_CHANNEL = '709405505990426624' // #staff-botspam
+const MUTED_ROLE = '710120710084755477' // Muted
+
+
+const recently_posted = {
+    /* 'userid': [ recent messages, queue with length MESSAGES_TO_CACHE ] */
+}
+
+module.exports = async message => {
+	if (message.author.bot || IGNORED_CHANNELS.includes(message.channel.id)
+                           || !ACTIVE_GUILDS.includes(message.guild.id))
+		return;
+
+    // initialize message cache for user
+    if (!(message.author.id in recently_posted))
+        recently_posted[message.author.id] = []
```

Comment:
I am confused, shouldn't this be on the same line? Either way there should be a semicolon here:

```suggestion
    if (!(message.author.id in recently_posted))
        recently_posted[message.author.id] = [];
```

Reply:
Been working too much with Python recently, I've fixed the variable names and lack of semicolons.